Objective-C/CBLCoreMLPredictiveModel.m

```diff
@@ -0,0 +1,538 @@
+    if (observations.count == 0)
+        return nil;
+    
+    // See https://developer.apple.com/documentation/vision/vncoremlrequest for
+    // the information about the VNObservation results:
+    NSString* predictedFeatureName = _model.modelDescription.predictedFeatureName;
+    if (predictedFeatureName) {
+        CBLMutableDictionary* result = [[CBLMutableDictionary alloc] init];
+        
+        // Predicted Feature:
+        VNClassificationObservation* top = observations[0];
+        [result setValue: top.identifier forKey: predictedFeatureName];
+        
+        // Predicted Probabilities:
+        NSString* predictedProbsName = _model.modelDescription.predictedProbabilitiesName;
+        if (predictedProbsName) {
+            CBLMutableDictionary* probs = [[CBLMutableDictionary alloc] init];
+            int count = 0;
+            for (VNClassificationObservation* o in observations) {
+                [probs setFloat: o.confidence forKey: o.identifier];
+                if (++count == kCBLMaxVisionPredictedProbabilities)
+                    break;
+            }
+            [result setDictionary: probs forKey: predictedProbsName];
+        }
+        return result;
+    } else {
+        // Note: Only one output is currently supported as there are no ways to map the observation
+        // outputs (Array) to MLModel outputs (Dictionary) when the outputs are multiple.
+        id value = nil;
+        MLModelDescription* desc = _model.modelDescription;
+        NSArray<NSString*>* outputKeys = desc.outputDescriptionsByName.allKeys;
+        if (outputKeys.count == 1 && observations.count == 1) { // Make sure that there is only one
+            VNObservation* o = observations[0];
+            if ([o isKindOfClass: VNCoreMLFeatureValueObservation.class]) {
+                MLFeatureValue* featureValue = ((VNCoreMLFeatureValueObservation*)o).featureValue;
+                value = [self.class valueFromFeatureValue: featureValue];
+            } else if ([o isKindOfClass: VNPixelBufferObservation.class]) {
+                // Make sure output type is image
+                if (desc.outputDescriptionsByName.allValues[0].type == MLFeatureTypeImage) {
+                    CVPixelBufferRef pixel = ((VNPixelBufferObservation*)o).pixelBuffer;
+                    value = [self.class blobFromPixelBuffer: pixel];
+                }
+            }
+        }
+        
+        if (!value)
+            return nil;
+        
+        CBLMutableDictionary* result = [[CBLMutableDictionary alloc] init];
+        [result setValue: value forKey: outputKeys[0]];
+        return result;
+    }
+}
+
+
+#pragma mark - CBLDictionary to Features Conversion
+
+
++ (id<MLFeatureProvider>) featuresFromDictionary: (CBLDictionary*)dictionary
+                                            spec: (NSDictionary<NSString*, MLFeatureDescription*>*)spec {
+    NSMutableDictionary* dict = [[NSMutableDictionary alloc] initWithCapacity: spec.count];
```

Comment:
since the parameter we receive is also a `dictionary`, and this variable also is `dict` looks similar. may be `featureProviderDictionary`?? what do you think?

Reply:
How about `featureProviderDict`?

Reply:
yea!